### PR TITLE
Votes grouping

### LINKED ISSUE #787: Copy of Multichoice polls votes grouping
On multichoice polls where it shows the votes, when a voter has selected more than one option, we need to change the display as follows:

Group the rows so that each person's vote shows as one section, with subsections within each vote showing which options the user voted for.

### PR DESCRIPTION
**Pull request by fabiolalombardim**
Closes #787 

Captura de pantalla 2024-04-14 a las 4 32 08

### CODE REVIEW

**Comment by fabiolalombardim** [comment]
Captura de pantalla 2024-04-14 a las 12 08 20
Css added & responsive table

**Comment by fabiolalombardim** [comment]
Captura de pantalla 2024-04-27 a las 11 39 56

Adding grouping solution to CSV file @ashutoshpw @EightRice @Man-Jain

File: package.json [hunk 1]
```diff
@@ -104,7 +104,7 @@
     "lint-staged": "15.2.2",
     "prettier": "^2.2.0",
     "source-map-explorer": "2.5.2",
-    "typescript": "^5.0.4"
+    "typescript": "5.4.0-beta"
```
Comment (Man-Jain) [inline package.json:107]: We shouldn't be using a beta version of typescript.

File: src/modules/lite/explorer/components/VoteDetails.tsx [hunk 2]
```diff
@@ -218,8 +220,10 @@ export const VoteDetails: React.FC<{
             )}
             {getTotalVoters(choices) > 0 ? (
               <DownloadCsvFile
-                data={choices}
+                data={groupedVotes}
                 pollId={poll?._id}
+                decimals={tokenData?.decimals ? tokenData?.decimals : ""}
```
Comment (Man-Jain) [inline src/modules/lite/explorer/components/VoteDetails.tsx:225]: should there be a isXTZ check here for 6 decimals ?

**Review by Man-Jain — COMMENTED** [review]
Only 2 little comments. Rest ready to merge

File: src/modules/lite/explorer/components/VoteDetails.tsx [hunk 3]
```diff
@@ -228,7 +232,8 @@ export const VoteDetails: React.FC<{
         <VotesDialog
           decimals={tokenData?.decimals ? tokenData?.decimals : ""}
```
Comment (Man-Jain) [inline src/modules/lite/explorer/components/VoteDetails.tsx:233]: Same here

**Review by Man-Jain — CHANGES_REQUESTED** [review]
Can be merged after the decimal fix

File: src/modules/lite/explorer/components/VoteDetails.tsx (hunk 2 above)
Comment (Man-Jain) [reply src/modules/lite/explorer/components/VoteDetails.tsx:225]: @fabiolalombardim Do we not need the decimal change here too ?

```decimals={tokenData?.decimals ? tokenData?.decimals : "6"```

File: src/modules/lite/explorer/components/VoteDetails.tsx [hunk 4]
```diff
@@ -218,17 +220,20 @@ export const VoteDetails: React.FC<{
...
+                decimals={tokenData?.decimals ? tokenData?.decimals : ""}
+                isXTZ={isXTZ}
                 symbol={isXTZ ? "XTZ" : tokenData?.symbol ? tokenData?.symbol : ""}
               />
             ) : null}
           </Grid>
         </LegendContainer>
         <VotesDialog
-          decimals={tokenData?.decimals ? tokenData?.decimals : ""}
+          decimals={tokenData?.decimals ? tokenData?.decimals : "6"}
```
Comment (Man-Jain) [inline src/modules/lite/explorer/components/VoteDetails.tsx:233]: I'm not sure if it's the best idea to default the decimal to 6, I'm not sure what we can do here but decimals = 6 should only be the case with isXTZ

**Review by Man-Jain — APPROVED** [review]

**Pull request merged**